### PR TITLE
Separate business logic from http layer

### PR DESCRIPTION
**Pull request by velzevur**
* Refactored the serialization to be separated from the node internals

* Moved business logic to a separate module `aehttp_logic`

* Moved signed transaction-specific logic (that is going to be deprecated) to a `aehttp_int_tx_logic module`

Still TODO: rebase on top of noise protocol changes and apply them accordingly

### CODE REVIEW

File: apps/aehttp/src/aehttp_int_tx_logic.erl [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                        expires_at => aeo_oracles:expires(O)}
+            end,
+            Oracles),
+    {ok, FmtOracles}.
+
+get_oracle_questions(OracleId, From, Max) ->
+    {ok, Queries} = aec_chain:get_open_oracle_queries(OracleId, From, Max),
+    FmtQueries =
+        lists:map(
+            fun(Q) -> #{<<"query_id">> => aeo_query:id(Q),
```
Comment (gorillainduction) [inline apps/aehttp/src/aehttp_int_tx_logic.erl]: Should this be a binary string?

File: apps/aehttp/src/aehttp_int_tx_logic.erl [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+                query_id => DecodedQueryId,
+                response => Response,
+                fee      => Fee})
+          end).
+
+get_oracles(From, Max) ->
+    {ok, Oracles} = aec_chain:get_oracles(From, Max),
+    FmtOracles =
+        lists:map(
+            fun(O) -> #{<<"address">> => aeo_oracles:id(O),
```
Comment (gorillainduction) [inline apps/aehttp/src/aehttp_int_tx_logic.erl]: Should this be a binary string?

File: apps/aehttp/src/aehttp_int_tx_logic.erl (hunk 1 above)
Comment (velzevur) [reply apps/aehttp/src/aehttp_int_tx_logic.erl]: It is encoded later on

**Comment by velzevur** [comment]
Rebased on top of master and incorporated `Noise` changes

File: apps/aecore/src/aec_peer_connection.erl [hunk 3]
```diff
@@ -541,7 +541,7 @@ get_header(height, N) ->
 get_header(Fun, Arg) ->
     case Fun(Arg) of
         {ok, Header} ->
-            {ok, HH} = aec_headers:serialize_to_map(Header),
+            HH = aehttp_api_parser:encode(header, Header),
```
Comment (lucafavatella) [inline apps/aecore/src/aec_peer_connection.erl]: I see this keeps base58. Ok.

[the remark this answers is not in the document]
Comment (velzevur) [reply apps/aecore/src/aec_peer_connection.erl]: I think all HTTP objects should not be part of the aec_peer_connection at all. So there should be a follow up PR fixing this.

[the remark this answers is not in the document]
Comment (lucafavatella) [reply apps/aecore/src/aec_peer_connection.erl]: So I understand idea is that complex items are serialized in distinct module, while hashes can get base58c encoded where they are. Makes sense.

**Pull request merged**